units/cloud_cleanup.go

```diff
@@ -67,21 +67,33 @@ func (j *cloudCleanupJob) Run(ctx context.Context) {
 		j.env = evergreen.GetEnvironment()
 	}
 
-	cloudManager, err := cloud.GetManager(ctx, j.env, cloud.ManagerOpts{
-		Provider: j.Provider,
-		Region:   j.Region,
+	accountRoles := j.env.Settings().Providers.AWS.AccountRoles
+	//Adding the empty value account as internally empty string is mapped to Kernel-Build AWS account
+	accountRoles = append(accountRoles, evergreen.AWSAccountRoleMapping{
+		Account: "",
 	})
-	if err != nil {
-		j.AddError(errors.Wrapf(err, "getting cloud manager for provider '%s' in region '%s'", j.Provider, j.Region))
-		return
-	}
 
-	err = cloudManager.Cleanup(ctx)
-	j.AddError(errors.Wrap(err, "cleaning up for provider"))
-	grip.Error(message.WrapError(err, message.Fields{
-		"message":  "cleaning up cloud resources",
-		"provider": j.Provider,
-		"region":   j.Region,
-		"job_id":   j.ID(),
-	}))
+	for _, accountRole := range accountRoles {
+		grip.Info(fmt.Sprintf("starting clean up for provider with account '%s'", accountRole.Account))
+
+		cloudManager, err := cloud.GetManager(ctx, j.env, cloud.ManagerOpts{
+			Provider: j.Provider,
+			Region:   j.Region,
+			Account:  accountRole.Account,
+		})
+
+		if err != nil {
+			j.AddError(errors.Wrapf(err, "getting cloud manager for provider '%s' in region '%s'", j.Provider, j.Region))
+			return
```

Comment:
I would prefer to continue here so we can at least try to clean up some of the other resources if we encounter an error